### DISCUSSION TITLE
astral issue blocking conda-forge

### DISCUSSION THREAD

**Discussion by callumrollo** [GliderToolsCommunity/GliderTools#56] [General]
In Issue #45, we've hit a block in getting GliderTools onto conda-forge due to the astral package having a different version on conda-forge than Python. Another scientific tool developer suggested moving to skyfield to get around this problem https://github.com/conda-forge/staged-recipes/pull/14282#issuecomment-802453894

Currently, astral is only used in one function of the optics.py script. I don't think it would be a lot of work to rewrite in skyfield. Is this worth persuing?

**Comment by isgiddy** [comment]
I had a quick look at skyfield, looks pretty similar to Astral. I am happy to work on this, I've been fixing a bug in this code anyway.

**Reply by jbusecke** [reply]
Great! Thank you for your efforts @isgiddy! Once you think you have a working implementation I can modify #45

**Reply by tjryankeogh** [reply]
Would be worthwhile confirming how skyfield handles positions at the poles during winter/summer, i.e. when no sunset/sunrise occurs. Right now Astral gives errors.

**Comment by isgiddy** [comment]
Skyfield actually deals with this in a  bit more use friendly way -
outputting polar day or polar night, I edited the code to therefore set
this to, for example polar day: 00:00 | 23:59 or polar night: 11:59 | 12:00
. Does that make sense?

**Reply by tjryankeogh** [reply]
Yep this makes sense. 
Because of polar day deployments, some time will need to be spent in integrating other quenching correction methods as options. As the current method does not work unless there are night time profiles.
I think the next best gold standard quenching correction method should be that of Xing et al. 2018. It was developed for Argos but I have tested it on Gliders and CTD Fluorescence profiles and find that it works really. It does require a PAR sensor though.
Meaning we need to have a third method where there is no PAR sensor and gliders are deployed during Polar day.

**Reply by bastienqueste** [reply]
Xing et al 2018 was a treat to use on the Oman data. Worked super well. Glad to see you found the same.
